public/app/features/alerting/unified/components/rule-editor/useDashboardQuery.ts

```diff
@@ -0,0 +1,27 @@
+import memoizeOne from 'memoize-one';
+
+import { DashboardDTO } from '../../../../../types';
+import { DashboardModel } from '../../../../dashboard/state';
+import { dashboardApi } from '../../api/dashboardApi';
+
+const convertToDashboardModel = memoizeOne((dashboardDTO: DashboardDTO) => {
```

Comment:
Is there a particular reason why we want to memoize (the last invocation) here?

Reply:
Yes. If I understand the docs correctly, `selectFromResult` does referential comparison. As the function we use in `selectFromResult` creates a new `DashboardModel` all the time, it would cause many unnecessary rerenders and recreation of `DashboardModel` object